File: src/models/projects.py
from src.extensions import db
from src.models.base import BaseModel

class Projects(db.Model, BaseModel):
    __tablename__ = "project"

    id = db.Column(db.Integer, primary_key=True)
    projects_name = db.Column(db.String, nullable=False)
    contract_number = db.Column(db.String, nullable=True)
    start_time = db.Column(db.Date, nullable=False)
    end_time = db.Column(db.Date, nullable=False)
    contractor = db.Column(db.String, nullable=True)
    proj_location = db.Column(db.String, nullable=False)
    proj_latitude = db.Column(db.Float, nullable=False)
    proj_longitude = db.Column(db.Float, nullable=False)
    geological_study = db.Column(db.Boolean, nullable=False, default=False)
    geophysical_study = db.Column(db.Boolean, nullable=False, default=False)
    hazard_study = db.Column(db.Boolean, nullable=False, default=False)
    geodetic_study = db.Column(db.Boolean, nullable=False, default=False)
    other_study = db.Column(db.Boolean, nullable=False, default=False)

    geological = db.relationship('Geological', back_populates='project')
    geophysical = db.relationship('Geophysical', back_populates='project')

    def __repr__(self):
        return f'<Projects {self.id} {self.projects_name}>'

File: src/static/js/projects.js
document.addEventListener("DOMContentLoaded", function() {
    // Fetch data from API endpoint
    fetch('/api/projects')
        .then(response => response.json())
        .then(data => {

            const projectTableBody = document.getElementById('projectTableBody');
            data.forEach(project => {
                const row = `
                    <tr>
                        <td>${project.projects_name}</td>
                        <td>${project.contract_number}</td>
                        <td>${project.start_time}</td>
                        <td>${project.end_time}</td>
                        <td>${project.contractor}</td>
                        <td>${project.proj_location}</td>
                        <td>${project.proj_latitude}</td>
                        <td>${project.proj_longitude}</td>
                        <td>${project.geological_study ? "Yes" : "No"}</td>
                        <td>${project.geophysical_study ? "Yes" : "No"}</td>
                        <td>${project.hazard_study ? "Yes" : "No"}</td>
                        <td>${project.geodetic_study ? "Yes" : "No"}</td>
                        <td>${project.other_study ? "Yes" : "No"}</td>
                        <td>
                            <a class="btn btn-info" href="/edit_project/${project.id}">Edit</a>
                        </td>
                        <td>
                            <form action="#" method="POST" style="display:inline;">
                                <button type="submit" class="btn btn-danger btn-block" onclick="return confirm('Are you sure you want to delete this station?');">Delete</button>
                            </form>
                        </td>
                    </tr>
                `;
                projectTableBody.innerHTML += row;
            });
        })
        .catch(error => {
            console.error('Error fetching data:', error);
            // Handle error scenario, e.g., show an error message on the UI
        });
});